### PR TITLE
add solution in clojure

### PR DESCRIPTION
**Pull request by mmcdon20**
## Description

Adds a solution in the [clojure programming language](https://clojure.org/).

This is a faithful 1-bit implementation.

Just like my umple and groovy solutions, this makes use of a java `BitSet` to store the prime flags.

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by mmcdon20** [comment]
This PR is stable/ready for review.

**Comment by rbergen** [comment]
@mmcdon20 I'm aware that Clojure is a JVM language and smooth interop with Java is very much a characteristic of the language. However, due to the nature of and philosophy behind the language, I would love to see a "full Clojure" implementation, too. Is this something you would entertain adding?

**Comment by mmcdon20** [comment]
@rbergen I'm not an expert in the language, but it looks like someone else just submitted another clojure solution (#605). I'll leave the "full Clojure" implementation to @axvr.

**Comment by rbergen** [comment]
Yes, #605 being opened is extremely convenient in the context of my question.

Maybe I'll casually mention that I would also really like to add a solution in GNU APL, and see if that works, too. :)

File: PrimeClojure/solution_1/PrimeClojure.clj [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                (if (< num sieveSize) 
+                    (recur (+ num (* factor 2)))))
+            
+            (def factor (+ factor 2))))
+
+    (printResults [this showResults duration passes]
+        (if showResults 
+            (printf "2, "))
+        
+        (def count (if (>= sieveSize 2) 1 0))
```
Comment (rbergen) [inline PrimeClojure/solution_1/PrimeClojure.clj]: I believe this def generates a warning at runtime:
```
WARNING: count already refers to: #'clojure.core/count in namespace: user, being replaced by: #'user/count
``` 

The solution still works, so I won't withhold approval over this. It's up to you if you want to do something about it.

**Review by rbergen — APPROVED** [review]
LGTM, with one comment

File: PrimeClojure/solution_1/PrimeClojure.clj (hunk 1 above)
Comment (mmcdon20) [reply PrimeClojure/solution_1/PrimeClojure.clj]: The count variable was shadowing a function in the standard library. I just renamed the variable to total.

**Review by marghidanu — APPROVED** [review]

**Pull request merged**